Makefile

```diff
@@ -26,40 +30,75 @@ install:
                 "* only serve: make serve \n" \
                 "* clean built doc files: make clean-doc \n" \
                 "* clean full environment: make clean \n" \
-		"* check spelling: make spelling \n" \
+                "* check spelling: make spelling \n" \
                 "* check inclusive language: make woke \n" \
+                "* other possible targets: make <press TAB twice> \n" \
 		"--------------------------------------------------------------- \n"
-run:
+
+
+install: $(VENVDIR)
+
+.PHONY:  install
+
+
+run: install
 	. $(VENV); sphinx-autobuild -c . -b dirhtml "$(SOURCEDIR)" "$(BUILDDIR)"
 
-html:
+.PHONY: run
+
+# Doesn't depend on $(BUILDDIR) to rebuild properly at every run
+html: install
 	. $(VENV); $(SPHINXBUILD) -c . -b dirhtml "$(SOURCEDIR)" "$(BUILDDIR)" -w .sphinx/warnings.txt
 
-epub:
+.PHONY: html
+
+
+epub: install
 	. $(VENV); $(SPHINXBUILD) -c . -b epub "$(SOURCEDIR)" "$(BUILDDIR)" -w .sphinx/warnings.txt
 
-serve:
+.PHONY: epub
+
+
+serve: html
 	cd "$(BUILDDIR)"; python3 -m http.server 8000
 
+.PHONY: serve
+
+
 clean: clean-doc
-	rm -rf .sphinx/venv
+	rm -rf $(VENVDIR)
```

Comment:
The explicit `.sphinx/venv` is actually a safety-net. I once accidentally changed or deleted a directory variable, ran `make clean` - and watched it merrily doing an `rm -rf /` on my hard disk. 

Is there a way we can avoid that?

Reply:
I'd be happy with that check. It's hard to make things foolproof with so many ingenious fools around.